### PR TITLE
Handle environment section in map format

### LINKED ISSUE #112: Handle environment section in map format
If you have an environment section like:

```
environment: 
  FOO: bar
  BAZ: qux
```

it will cause failures with the following error:

```
invalid manifest: yaml: unmarshal errors: line 4: cannot unmarshal !!map into []string
```

### PR DESCRIPTION
**Pull request by nathany**
## Release Playbook
- [ ] Rebase against master
- [ ] Release branch ()
- [ ] Pass CI ()
- [ ] Code review
- [ ] Merge into master
- [ ] Release master ()
- [ ] Pass CI ()
- [ ] Update staging rack
- [ ] Edit [Rack release record](https://github.com/convox/rack/releases) and/or update [docs](https://github.com/convox/convox.github.io)
- [ ] Publish release
- [ ] Release CLI

fixes #112

this pulls in some code from libcompose (Apache license) but modifies it to work with gopkg.in/yaml.v2.

### CODE REVIEW

**Comment by awsmsrc** [comment]
We should consider duplicating this behavior in manifest/manifest.go as well, and possibly porting the tests from #576. As ideally we will be losing this model sooner rather then later and this will also prevent sending a bad environment in the first place.

I'm unsure how that would effect YAML map merging however.

File: api/models/manifest.go [hunk 1]
```diff
@@ -532,3 +532,69 @@ func (me ManifestEntry) RegistryImage(app *App, buildId string) string {
 
 	return fmt.Sprintf("%s/%s-%s:%s", os.Getenv("REGISTRY_HOST"), app.Name, me.Name, buildId)
 }
+
+type MapOrEqualSlice []string
+
+// UnmarshalYAML implements the Unmarshaller interface.
+func (s *MapOrEqualSlice) UnmarshalYAML(unmarshal func(interface{}) error) error {
```
Comment (nathany) [inline api/models/manifest.go]: this is the only interesting bit, changing the Unmarshaller interface from what libcompose does to what gopkg.in does.

**Comment by ddollar** [comment]
:+1: Looks good. Thanks for adding the test, too!

**Comment by mattmanning** [comment]
Merged in #584

**Pull request closed without merging**